### PR TITLE
#810 - EEZ report

### PR DESCRIPTION
**Pull request by andres-arana**
Connects https://github.com/SkyTruth/pelagos-server/issues/810.

This is the first part of the EEZ report implementation. It contains code to process some configuration fields in a layer and display a generate report button on the info window for the layer. The button when clicked displays a dialog which shows a prompt and contains some controls to further refine the selection.

I'm not actually posting the request to the server because I'm doing that in another PR later, as I want to check that with the server before committing the code.

### CODE REVIEW

File: js/app/Data/CartoDBInfoWindow.js [hunk 1]
```diff
@@ -78,7 +77,9 @@ define([
          obj
         );
 
-        cb($(popupHtml).find('.cartodb-popup-content'));
+        var html = $(popupHtml).find('.cartodb-popup-content');
+
+        cb(html, data);
```
Comment (redhog) [inline js/app/Data/CartoDBInfoWindow.js:82]: This is confusing cb with two args is usually cb(err, data).
Is there a big reason you didn't do 

```
obj.html = html;
obj.toString = function () { return this.html);
return obj;
```

? This is a pattern I've used in many places...

File: js/app/Visualization/Animation/AnimationManager.js [hunk 2]
```diff
@@ -306,6 +306,26 @@ function(Class,
...
+          animation.args &&
+          animation.args.source &&
+          animation.args.source.args &&
+          animation.args.source.args.url;
+
+        var isReportable =
+          animation.args &&
+          animation.args.reportable
+
+        return hasSourceUrl && isReportable;
```
Comment (redhog) [inline js/app/Visualization/Animation/AnimationManager.js:323]: Why check source? You don't trust args.reportable?

File: docs/schema.md [hunk 3]
```diff
@@ -13,317 +19,465 @@ This schema document describes the schemas for each part in this stack:
...
+          "label": "EEZ"
         },
 
-        /* Overlay animation confirguration. See below. */
-        "animations": []
-      }
-    }
+        /* Here we are publishing another field from the polygon, in this case one that
+           is directly available in the templates */
+        "country": {}
```
Comment (redhog) [inline docs/schema.md:308]: Why is this needed at all / why are not all fields from the polygon available to the templates (with the titles as names)?

File: js/app/Data/CartoDBInfoWindow.js (hunk 1 above)
Comment (andres-arana) [inline js/app/Data/CartoDBInfoWindow.js:82]: You are absolutely right. I'll change it so that the result includes both the html and the data as fields instead, and make it a single-arg callback again.

File: js/app/Visualization/Animation/AnimationManager.js (hunk 2 above)
Comment (andres-arana) [inline js/app/Visualization/Animation/AnimationManager.js:323]: I'm still not 100% sure how the rest of the code works, so I'm usually much more defensive than usual. That's why I tend to do these kind of things in the interface with the rest of the code.

But if `args.reportable` works, I'd use that. I'll try it out when I add the missing changes for the report generation.

File: docs/schema.md (hunk 3 above)
Comment (andres-arana) [inline docs/schema.md:308]: Implementation detail, it was easier this way and I needed it working quickly. I'll change it to copy all the fields from the data once I merge the stuff that has server changes.
